core/scripts/common/polygonedge.go

```diff
@@ -0,0 +1,184 @@
+package common
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/umbracle/fastrlp"
+)
+
+type Nonce [8]byte
+
+var (
+	nonceT = reflect.TypeOf(Nonce{})
+)
+
+func (n Nonce) String() string {
+	return EncodeToHex(n[:])
+}
+
+// MarshalText implements encoding.TextMarshaler
+func (n Nonce) MarshalText() ([]byte, error) {
+	return []byte(n.String()), nil
+}
+
+// UnmarshalJSON parses a nonce in hex syntax.
+func (n *Nonce) UnmarshalJSON(input []byte) error {
+	return hexutil.UnmarshalFixedJSON(nonceT, input, n[:])
+}
+
+type ExtraData string
+
+func (e ExtraData) Decode() ([]byte, error) {
+	return DecodeHex(string(e))
+}
+
+// Header represents a block header in the Ethereum blockchain.
+type PolygonEdgeHeader struct {
+	ParentHash   common.Hash    `json:"parentHash"       gencodec:"required"`
+	Sha3Uncles   common.Hash    `json:"sha3Uncles"       gencodec:"required"`
+	Miner        common.Address `json:"miner"`
+	StateRoot    common.Hash    `json:"stateRoot"        gencodec:"required"`
+	TxRoot       common.Hash    `json:"transactionsRoot" gencodec:"required"`
+	ReceiptsRoot common.Hash    `json:"receiptsRoot"     gencodec:"required"`
+	LogsBloom    types.Bloom    `json:"logsBloom"        gencodec:"required"`
+	Difficulty   hexutil.Uint64 `json:"difficulty"       gencodec:"required"`
+	Number       hexutil.Uint64 `json:"number"           gencodec:"required"`
+	GasLimit     hexutil.Uint64 `json:"gasLimit"         gencodec:"required"`
+	GasUsed      hexutil.Uint64 `json:"gasUsed"          gencodec:"required"`
+	Timestamp    hexutil.Uint64 `json:"timestamp"        gencodec:"required"`
+	ExtraData    ExtraData      `json:"extraData"        gencodec:"required"`
+	MixHash      common.Hash    `json:"mixHash"`
+	Nonce        Nonce          `json:"nonce"`
+	Hash         common.Hash    `json:"hash"`
+
+	// BaseFee was added by EIP-1559 and is ignored in legacy headers.
+	BaseFee    hexutil.Uint64 `json:"baseFeePerGas"`
+	BaseFeeAlt hexutil.Uint64 `json:"baseFee,omitempty"`
```

Comment:
Can you document this and why its there in addition to `BaseFee`?

Reply:
Sure